git_machete/client.py

```diff
@@ -2362,3 +2386,23 @@ def __sync_before_creating_pr(self, *, opt_onto: Optional[LocalBranchShortName],
             if ans in ('y', 'yes'):
                 return
             raise MacheteException('Pull request creation interrupted.')
+
+    def delete_untracked(self,
+                         opt_yes: bool,
+                         opt_merge: bool,
+                         opt_no_interactive_rebase: bool,
+                         opt_no_edit_merge: bool) -> None:
+        untracked_branches: List[LocalBranchShortName] = []
+        for managed_branch in self.managed_branches:
+            status, _ = self.__git.get_strict_remote_sync_status(managed_branch)
+            if status == SyncToRemoteStatuses.UNTRACKED:
```

Comment:
If there's no remote configured in the repo, no branch is removed, right?

Reply:
Yeah, no **untracked** branch is removed, but **unmanaged** branches are still getting removed by `machete_client.delete_unmanaged(opt_yes=cli_opts.opt_yes)` in cli.py